test/logger/instantiation.test.js

```diff
@@ -0,0 +1,339 @@
+'use strict'
+
+const stream = require('node:stream')
+const os = require('node:os')
+const fs = require('node:fs')
+
+const t = require('tap')
+const split = require('split2')
+
+const { streamSym } = require('pino/lib/symbols')
+
+const Fastify = require('../../fastify')
+const helper = require('../helper')
+const { FST_ERR_LOG_INVALID_LOGGER } = require('../../lib/errors')
+const { once, on } = stream
+const { createTempFile, request } = require('./loggerTestUtils')
+
+t.test('logger instantiation', (t) => {
+  t.setTimeout(60000)
+
+  let localhost
+  let localhostForURL
+
+  t.plan(11)
+  t.before(async function () {
+    [localhost, localhostForURL] = await helper.getLoopbackHost()
+  })
+
+  t.test('can use external logger instance', async (t) => {
+    const lines = [/^Server listening at /, /^incoming request$/, /^log success$/, /^request completed$/]
+    t.plan(lines.length + 1)
+
+    const stream = split(JSON.parse)
+
+    const logger = require('pino')(stream)
+
+    const fastify = Fastify({ logger })
+    t.teardown(fastify.close.bind(fastify))
+
+    fastify.get('/foo', function (req, reply) {
+      t.ok(req.log)
+      req.log.info('log success')
+      reply.send({ hello: 'world' })
+    })
+
+    await fastify.ready()
+    await fastify.listen({ port: 0, host: localhost })
```

Comment:
minor optimization - we can remove the ready call

```suggestion
    await fastify.listen({ port: 0, host: localhost })
```